File: frontend/desktop/lib/models/requests/delete_admin_request.dart
/// Body for DELETE /admins/{id}. Reason/RecipientEmail are only required by
/// the backend when the target is an OrganizationAdmin (see
/// AdminService.DeleteAsync) — always send this, even with both fields
/// null, since the route no longer accepts a bodyless DELETE.
class DeleteAdminRequest {
  final String? reason;
  final String? recipientEmail;

  const DeleteAdminRequest({this.reason, this.recipientEmail});

  Map<String, dynamic> toJson() => {
        if (reason != null && reason!.isNotEmpty) 'Reason': reason,
        if (recipientEmail != null && recipientEmail!.isNotEmpty) 'RecipientEmail': recipientEmail,
      };
}

File: backend/services/identity/eTicketing.Identity.Business/Admins/Dtos/DeleteAdminRequest.cs
namespace eTicketing.Identity.Business.Admins;

/// <summary>Body for DELETE /admins/{id}. Reason/RecipientEmail are only required when the
/// target is an OrganizationAdmin (enforced in AdminService.DeleteAsync, which is the only place
/// that knows the target's role) — always send this body, even empty, since the route no longer
/// accepts a bare DELETE.</summary>
public sealed record DeleteAdminRequest
{
    public string? Reason { get; init; }
    public string? RecipientEmail { get; init; }
}
